### PR TITLE
HDDS-11623. Improve OM Ratis Configuration change log message

### PR DESCRIPTION
**Pull request by harryteng9527**
## What changes were proposed in this pull request?
This PR removes the `startupRole` from the peer list log in `OzoneManagerStateMachine` to avoid confusing others.

Here are the logs before and after the modification.
### Before change
```
Received Configuration change notification from Ratis. New Peer list:
[id: "omNode-3"
address: "localhost:15015"
startupRole: FOLLOWER
, id: "omNode-1"
address: "localhost:15007"
startupRole: FOLLOWER
, id: "omNode-2"
address: "localhost:15011"
startupRole: FOLLOWER
]
```
### After change
```
Received Configuration change notification from Ratis. New Peer list:
[id: "omNode-3"
 address: "localhost:15015"
, id: "omNode-1"
 address: "localhost:15007"
, id: "omNode-2"
 address: "localhost:15011"
]
```

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-11623

## How was this patch tested?

Pass all tests.

### CODE REVIEW

**Review by sarvekshayr — COMMENTED** [review]
@harryteng9527 Thank you for working on this patch. LGTM.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java [hunk 1]
```diff
@@ -216,7 +217,9 @@ public void notifyConfigurationChanged(long term, long index,
     List<RaftProtos.RaftPeerProto> newPeers =
         newRaftConfiguration.getPeersList();
     LOG.info("Received Configuration change notification from Ratis. New Peer" +
-        " list:\n{}", newPeers);
+        " list:\n[{}\n]", newPeers.stream().map(peer ->
+            String.format("id: \"%s\"%n address: \"%s\"", peer.getId().toStringUtf8(), peer.getAddress()))
+        .collect(Collectors.joining("\n, ")));
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java:222]: I would prefer to format the string without newlines.  Log messages split into several lines are prone to be mixed up with messages from other threads.

Also, would be nice to move the processing out of the log statement.

```suggestion
    List<String> formatted = newPeers.stream()
        .map(peer -> String.format("id: \"%s\" address: \"%s\"", peer.getId().toStringUtf8(), peer.getAddress()))
        .collect(Collectors.toList());
    LOG.info("Received Configuration change notification from Ratis. New Peer list: {}", formatted);
```

Comment (harryteng9527) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java]: I originally intended to maintain the original log style. Thanks for reviewing!

After the modification, the current log：
```=bash
Received Configuration change notification from Ratis. New Peer list: [id: "omNode-3" address: "localhost:15015", id: "omNode-1" address: "localhost:15007", id: "omNode-2" address: "localhost:15011"]
```

Comment (adoroszlai) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java]: Thanks @harryteng9527 for updating the patch.

**Comment by szetszwo** [comment]
@adoroszlai , thanks for the reminder.  Will review this.

**Review by szetszwo — COMMENTED** [review]
@harryteng9527 , thanks a lot for working on this!  Please see the comment inlined.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java [hunk 2]
```diff
@@ -215,8 +216,10 @@ public void notifyConfigurationChanged(long term, long index,
       RaftProtos.RaftConfigurationProto newRaftConfiguration) {
     List<RaftProtos.RaftPeerProto> newPeers =
         newRaftConfiguration.getPeersList();
-    LOG.info("Received Configuration change notification from Ratis. New Peer" +
-        " list:\n{}", newPeers);
+    List<String> newPeersLogWithoutStartupRole = newPeers.stream()
+        .map(peer -> String.format("id: \"%s\" address: \"%s\"", peer.getId().toStringUtf8(), peer.getAddress()))
+        .collect(Collectors.toList());
+    LOG.info("Received Configuration change notification from Ratis. New Peer list: {}", newPeersLogWithoutStartupRole);
```
[the remark this answers is not in the document]
Comment (harryteng9527) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java:222]: Thanks for reviewing! 

After the modification, the current log will be like：
```=bash
notifyConfigurationChanged from Ratis: term=1, index=0, New Peer list: omNode-3(localhost:15015), omNode-1(localhost:15007), omNode-2(localhost:15011)
```

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by adoroszlai** [comment]
Thanks @harryteng9527 for the patch, @sarvekshayr, @szetszwo for the review.

**Pull request merged**